File: utils/src/main/java/com/example/utils/ClipboardUtil.java
package com.example.utils;

import android.content.ClipData;
import android.content.ClipDescription;
import android.content.ClipboardManager;
import android.content.Context;
import android.content.Intent;
import android.net.Uri;

/**
 * @author fqxyi
 * @date 2020/8/20
 * 复制粘贴文本的三种类型
 */
public class ClipboardUtil {
    /**
     * 复制 类型一: text
     */
    public static void copy(Context context, String text) {
        ClipboardManager clipboard = (ClipboardManager) context.getSystemService(Context.CLIPBOARD_SERVICE);
        ClipData textCd = ClipData.newPlainText("text", text);
        clipboard.setPrimaryClip(textCd);
    }

    /**
     * 粘贴 类型一: text
     */
    public static CharSequence pasteText(Context context) {
        ClipboardManager clipboard = (ClipboardManager) context.getSystemService(Context.CLIPBOARD_SERVICE);
        if (clipboard.getPrimaryClipDescription().hasMimeType(ClipDescription.MIMETYPE_TEXT_PLAIN)) {
            ClipData cdText = clipboard.getPrimaryClip();
            // 此处是 text
            ClipData.Item item = cdText.getItemAt(0);
            return item.getText();
        }
        return null;
    }

    /**
     * 复制 类型二: Uri
     */
    public static void copy(Context context, Uri uri) {
        ClipboardManager clipboard = (ClipboardManager) context.getSystemService(Context.CLIPBOARD_SERVICE);
        ClipData clipUri = ClipData.newUri(context.getContentResolver(), "Uri", uri);
        clipboard.setPrimaryClip(clipUri);
    }

    /**
     * 粘贴 类型二: Uri
     */
    public static Uri pasteUri(Context context) {
        ClipboardManager clipboard = (ClipboardManager) context.getSystemService(Context.CLIPBOARD_SERVICE);
        if (clipboard.getPrimaryClipDescription().hasMimeType(ClipDescription.MIMETYPE_TEXT_URILIST)) {
            ClipData cdUri = clipboard.getPrimaryClip();
            // 此处是 Uri
            ClipData.Item item = cdUri.getItemAt(0);
            return item.getUri();
        }
        return null;
    }

    /**
     * 复制 类型三: Intent
     */
    public static void copy(Context context, Intent intent) {
        ClipboardManager clipboard = (ClipboardManager) context.getSystemService(Context.CLIPBOARD_SERVICE);
        ClipData clipIntent = ClipData.newIntent("Intent", intent);
        clipboard.setPrimaryClip(clipIntent);
    }

    /**
     * 粘贴 类型三: Intent
     */
    public static Intent pasteIntent(Context context) {
        ClipboardManager clipboard = (ClipboardManager) context.getSystemService(Context.CLIPBOARD_SERVICE);
        if (clipboard.getPrimaryClipDescription().hasMimeType(ClipDescription.MIMETYPE_TEXT_INTENT)) {
            ClipData cdIntent = clipboard.getPrimaryClip();
            // 此处是 Intent
            ClipData.Item item = cdIntent.getItemAt(0);
            return item.getIntent();
        }
        return null;
    }
}

File: utils/src/main/java/com/example/utils/device/ClipboardUtil.kt
package com.example.utils.device

import android.content.*
import android.net.Uri

/**
 * @author fqxyi
 * @date 2020/8/20
 * 复制粘贴文本的三种类型
 */
object ClipboardUtil {

    /**
     * 复制 类型一: text
     */
    fun copy(context: Context, text: String?) {
        val clipboard = context.getSystemService(Context.CLIPBOARD_SERVICE) as ClipboardManager
        val textCd = ClipData.newPlainText("text", text)
        clipboard.setPrimaryClip(textCd)
    }

    /**
     * 粘贴 类型一: text
     */
    fun pasteText(context: Context): CharSequence? {
        val clipboard = context.getSystemService(Context.CLIPBOARD_SERVICE) as ClipboardManager
        if (clipboard.primaryClipDescription?.hasMimeType(ClipDescription.MIMETYPE_TEXT_PLAIN) == true) {
            return clipboard.primaryClip?.getItemAt(0)?.text
        }
        return null
    }

    /**
     * 复制 类型二: Uri
     */
    fun copy(context: Context, uri: Uri?) {
        val clipboard = context.getSystemService(Context.CLIPBOARD_SERVICE) as ClipboardManager
        val clipUri = ClipData.newUri(context.contentResolver, "Uri", uri)
        clipboard.setPrimaryClip(clipUri)
    }

    /**
     * 粘贴 类型二: Uri
     */
    fun pasteUri(context: Context): Uri? {
        val clipboard = context.getSystemService(Context.CLIPBOARD_SERVICE) as ClipboardManager
        if (clipboard.primaryClipDescription?.hasMimeType(ClipDescription.MIMETYPE_TEXT_URILIST) == true) {
            return clipboard.primaryClip?.getItemAt(0)?.uri
        }
        return null
    }

    /**
     * 复制 类型三: Intent
     */
    fun copy(context: Context, intent: Intent?) {
        val clipboard = context.getSystemService(Context.CLIPBOARD_SERVICE) as ClipboardManager
        val clipIntent = ClipData.newIntent("Intent", intent)
        clipboard.setPrimaryClip(clipIntent)
    }

    /**
     * 粘贴 类型三: Intent
     */
    fun pasteIntent(context: Context): Intent? {
        val clipboard = context.getSystemService(Context.CLIPBOARD_SERVICE) as ClipboardManager
        if (clipboard.primaryClipDescription?.hasMimeType(ClipDescription.MIMETYPE_TEXT_INTENT) == true) {
            return clipboard.primaryClip?.getItemAt(0)?.intent
        }
        return null
    }
}